services/book.services.js

```diff
@@ -111,14 +114,20 @@ async function createBorrowing(book, user_id, lastUpdatedBy, nosAvailable) {
         { where: { id: book.id } },
         { transaction }
       );
+    });
+
+    const allMessages = {
+      userData: userInfo,
+      bookData: book,
+      borrowedBook: borrow,
+    };
 
-      sendBorrowBookEmail(userInfo, book, borrow);
+    await producer.connect();
+    await producer.produce(allMessages);
 
-      return borrow;
-    });
+    return borrow;
   } catch (err) {
     logger.error(ERROR_MSG, err);
-
     return Promise.reject({ status: 500, msg: "Internal Server Error" });
```

Comment:
Try not to be giving user information about your system. You can sent back a default error message for error that's not specific and not internal server error. 

e.g `Unable to complete request at this time. Please try again.` You can still return the 500 http response code but don't return that message